soccer/vision/ball/BallBounce.cpp

```diff
@@ -0,0 +1,279 @@
+#include "BallBounce.hpp"
+
+#include <cmath>
+
+#include <Constants.hpp>
+
+#include "vision/util/VisionFilterConfig.hpp"
+
+REGISTER_CONFIGURABLE(BallBounce)
+
+ConfigDouble* BallBounce::robot_body_lin_dampen;
+ConfigDouble* BallBounce::robot_mouth_lin_dampen;
+ConfigDouble* BallBounce::robot_body_angle_dampen;
+ConfigDouble* BallBounce::robot_mouth_angle_dampen;
+
+/**
+ * Note 0 case returns -1 instead of 0
+ * Forced to check with small epislon since we actually care about 0 being represented
+ * correctly as -1. It's ok if the 1.0e-10 doesn't return -1 or 1.
+ */
+inline int sign(double val) { return (1.0e-10 < val) - (val <= 1.0e-10); }
+
+void BallBounce::createConfiguration(Configuration* cfg) {
+    robot_body_lin_dampen = new ConfigDouble(cfg, "VisionFilter/Bounce/robot_body_lin_dampen", .9);
+    robot_mouth_lin_dampen = new ConfigDouble(cfg, "VisionFilter/Bounce/robot_mouth_lin_dampen", .3);
+    robot_body_angle_dampen = new ConfigDouble(cfg, "VisionFilter/Bounce/robot_body_angle_dampen", 0);
+    robot_mouth_angle_dampen = new ConfigDouble(cfg, "VisionFilter/Bounce/robot_mouth_angle_dampen", 0);
+}
+
+bool BallBounce::CalcBallBounce(const KalmanBall& ball,
+                                const std::vector<WorldRobot>& yellowRobots,
+                                const std::vector<WorldRobot>& blueRobots,
+                                Geometry2d::Point& outNewVel) {
+    std::vector<WorldRobot> robots = yellowRobots;
```

Comment:
WorldRobot seems to be a fairly large type and you're making alot of copies. I would try to see if you can get rid of most of the unnecessary copies with refferences.
Do you really need copies of all the Robots here?
Couldn't this be a vector of const pointers? or a vector<std::reference_wrapper>

These copies seem to be happening alot rn.